src/main/java/capstone/mfslbackend/service/TransactionService.java

```diff
@@ -55,9 +62,56 @@ public Transaction createTransaction(Long fantasyTeamId, Long incomingPlayerId,
         }
 
         Player playerIn = playerService.getPlayerById(incomingPlayerId);
-        transaction.setPlayerIn(playerIn);
+        Player outgoingPlayer = playerService.getPlayerById(outgoingPlayerId);
+        String position = playerIn.getPosition();
+
+        if (playerIn.getTeam() == null) { //it means it is a free agent
 
-        transaction.setStatus(TransactionStatus.PROPOSED);
+            if (outgoingPlayer.getPosition().equals("Goalkeeper") && !(position.equals("Goalkeeper"))) {
```

Comment:
a fantasy team can have multiple goalies, but can only have 1 in the first 11 ids in the string playerIdsInOrder, so we're allowed to trade a goalie for a non-goalie